### PR TITLE
Fix assumption that arrays have 2+ dimensions in CF writer

### PR DESCRIPTION
**Pull request by pnuu**
The CF writer currently has an assumption that all arrays have both x and y coordinates. This might not be intentional, as the only place this is enforced is in a private method where coordinate units are check when trying to determine whether the data are projected or not.

This had come up in https://github.com/foua-pps/level1c4pps/ which uses Satpy to convert data to NWC SAF PPS compatible format.

The test I added is very simple, and without the `try/except` I added in this PR it'd fail with `KeyError: 'x'`.

Ping also to @ninahakansson 

 - [x] Closes https://github.com/foua-pps/level1c4pps/issues/74
 - [x] Tests added

### CODE REVIEW

**Review by mraspaud — APPROVED** [review]
LGTM, just one styling comment.

File: satpy/writers/cf_writer.py [hunk 1]
```diff
@@ -745,8 +745,12 @@ def _try_to_get_crs(new_data):
...
     def _try_get_units_from_coords(new_data):
         for c in "xy":
-            if "units" in new_data.coords[c].attrs:
-                return new_data.coords[c].attrs["units"]
+            try:
+                if "units" in new_data.coords[c].attrs:
+                    return new_data.coords[c].attrs["units"]
+            except KeyError:
+                # If the data has only 1 dimension, it has only one of x or y coords
+                pass
```
[the remark this answers is not in the document]
Comment (pnuu) [reply satpy/writers/cf_writer.py]: Ah, damn. I just pushed another simplification for this.

[the remark this answers is not in the document]
Comment (pnuu) [reply satpy/writers/cf_writer.py]: Used your version, it's cleaner.

**Comment by mraspaud** [comment]
3 tests failing now...

**Comment by pnuu** [comment]
The failing tests were because of the netCDF compression settings not working with the latest XArray version.

**Comment by pnuu** [comment]
I arranged the imports in `test_cf.py` so CodeScene complains about one thing less...

**Pull request merged**